### PR TITLE
Updates. DASH-143 link crypto handle updates

### PR DESCRIPTION
**Pull request by trukhilio**
- Create chain and token codes database tables
- Handle chain and token codes server side
- Handle chain and token codes client side
- Implement autocomplete chain and token codes fields
- Use autocomplete fields in link token list and request tokens
- UI fixes
- Check for results in EdgeConfirmAction fioExecutedAction

### CODE REVIEW

File: server/src/db.mjs [hunk 1]
```diff
@@ -2,53 +2,57 @@ import Sequelize from 'sequelize';
...
 PaymentEventLog.init(sequelize);
-AdminUsersRole.init(sequelize);
-AdminUsersStatus.init(sequelize);
+PublicWalletData.init(sequelize);
+ReferrerProfile.init(sequelize);
+TokenCode.init(sequelize);
+User.init(sequelize);
 Var.init(sequelize);
-AdminUser.init(sequelize);
+Wallet.init(sequelize);
```
Comment (andreyvEze) [inline server/src/db.mjs:55]: why is sorted

File: server/src/models/index.mjs [hunk 2]
```diff
@@ -3,24 +3,26 @@ import db from '../db';
...
   OrderItem,
+  OrderItemStatus,
   Payment,
   PaymentEventLog,
+  PublicWalletData,
+  ReferrerProfile,
+  TokenCode,
+  User,
   Var,
+  Wallet,
```
Comment (andreyvEze) [inline server/src/models/index.mjs:27]: could we remove the sort

File: server/src/routes/index.mjs [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
-  account,
-  refProfiles,
   newDeviceTwoFactor,
-  contacts,
+  notifications,
   orders,
   payments,
-  adminUsers,
+  refProfiles,
+  users,
```
Comment (andreyvEze) [inline server/src/routes/index.mjs:28]: could we remove sort changes?

File: client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    tokenCodeListHasOneItem,
+    tokenCodeFieldName,
+    selectedChainCodeId,
+    selectedChainCodeName,
+    fieldChange,
+  ]);
+
+  useEffectOnce(() => {
+    if (chainCodeInitialValue) {
+      handleChainCodeChange(chainCodeInitialValue);
```
Comment (andreyvEze) [inline client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx:90]: handleChainCodeChange should be in dependencies array I think

File: client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+            const { tokenCodeId: id, tokenCodeName: name } = chainCodeItem;
+            return { id, name };
+          })}
+          placeholder="Type or Select Token Code"
+          upperCased={true}
+          {...props}
+        />
+      </div>
+    </>
+  );
```
Comment (andreyvEze) [inline client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx:193]: why you did no create `useContext`?

File: client/src/pages/FioTokensRequestPage/components/RequestTokensForm/RequestTokensForm.tsx [hunk 6]
```diff
@@ -11,23 +11,30 @@ import AmountInput from '../../../../components/Input/AmountInput';
...
+
+import PublicKeyField from './PublicKeyField';
 
 import { formValidation, submitValidation } from './validation';
 import { minWaitTimeFunction } from '../../../../utils';
 import { fioAddressExistsValidator } from '../../../../util/validators';
 
 import { COLOR_TYPE } from '../../../../components/Input/ErrorBadge';
-import { CHAIN_CODE_LIST } from '../../../../constants/common';
+import { CHAIN_CODES } from '../../../../constants/common';
```
Comment (andreyvEze) [inline client/src/pages/FioTokensRequestPage/components/RequestTokensForm/RequestTokensForm.tsx:24]: should we remove CHAIN_CODE_LIST const?

Comment (trukhilio) [reply client/src/pages/FioTokensRequestPage/components/RequestTokensForm/RequestTokensForm.tsx:24]: yes, forgot about that

File: server/src/routes/index.mjs (hunk 3 above)
Comment (trukhilio) [reply server/src/routes/index.mjs:28]: yes

File: client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx (hunk 4 above)
Comment (trukhilio) [reply client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx]: done

File: client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx (hunk 5 above)
Comment (trukhilio) [reply client/src/components/ChainAndTokenCodesAutocompleteFields/ChainAndTokenCodesAutocompleteFields.tsx:84]: moved logic to useContext

File: client/src/pages/FioTokensRequestPage/components/RequestTokensForm/RequestTokensForm.tsx (hunk 6 above)
Comment (trukhilio) [reply client/src/pages/FioTokensRequestPage/components/RequestTokensForm/RequestTokensForm.tsx:24]: removed

**Pull request merged**